File: avalon-modules/avalon-authorization/src/main/java/io/github/lishangbu/avalon/authorization/controller/OauthRegisteredClientController.java
package io.github.lishangbu.avalon.authorization.controller;

import io.github.lishangbu.avalon.authorization.entity.OauthRegisteredClient;
import io.github.lishangbu.avalon.authorization.service.OauthRegisteredClientService;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/// OAuth2 注册客户端控制器
///
/// 提供 OAuth2 注册客户端管理相关接口
///
/// @author lishangbu
/// @since 2026/3/19
@RestController
@RequestMapping("/oauth-registered-client")
@RequiredArgsConstructor
public class OauthRegisteredClientController {

    private final OauthRegisteredClientService oauthRegisteredClientService;

    /// 分页条件查询注册客户端
    ///
    /// @param pageable         分页参数
    /// @param registeredClient 查询条件
    /// @return 注册客户端分页结果
    @GetMapping("/page")
    public Page<OauthRegisteredClient> getPage(
            Pageable pageable, OauthRegisteredClient registeredClient) {
        return oauthRegisteredClientService.getPageByCondition(registeredClient, pageable);
    }

    /// 条件查询注册客户端列表
    ///
    /// @param registeredClient 查询条件
    /// @return 注册客户端列表
    @GetMapping("/list")
    public List<OauthRegisteredClient> list(OauthRegisteredClient registeredClient) {
        return oauthRegisteredClientService.listByCondition(registeredClient);
    }

    /// 根据 ID 查询注册客户端
    ///
    /// @param id 主键
    /// @return 注册客户端
    @GetMapping("/{id}")
    public OauthRegisteredClient getById(@PathVariable String id) {
        return oauthRegisteredClientService.getById(id).orElse(null);
    }

    /// 新增注册客户端
    ///
    /// @param registeredClient 注册客户端实体
    /// @return 保存后的注册客户端
    @PostMapping
    public OauthRegisteredClient save(@RequestBody OauthRegisteredClient registeredClient) {
        return oauthRegisteredClientService.save(registeredClient);
    }

    /// 更新注册客户端
    ///
    /// @param registeredClient 注册客户端实体
    /// @return 更新后的注册客户端
    @PutMapping
    public OauthRegisteredClient update(@RequestBody OauthRegisteredClient registeredClient) {
        return oauthRegisteredClientService.update(registeredClient);
    }

    /// 根据 ID 删除注册客户端
    ///
    /// @param id 主键
    @DeleteMapping("/{id}")
    public void deleteById(@PathVariable String id) {
        oauthRegisteredClientService.removeById(id);
    }
}

File: avalon-modules/avalon-authorization/src/main/kotlin/io/github/lishangbu/avalon/authorization/controller/OauthRegisteredClientController.kt
package io.github.lishangbu.avalon.authorization.controller

import io.github.lishangbu.avalon.authorization.entity.dto.OauthRegisteredClientSpecification
import io.github.lishangbu.avalon.authorization.entity.dto.OauthRegisteredClientView
import io.github.lishangbu.avalon.authorization.entity.dto.SaveOauthRegisteredClientInput
import io.github.lishangbu.avalon.authorization.entity.dto.UpdateOauthRegisteredClientInput
import io.github.lishangbu.avalon.authorization.service.OauthRegisteredClientService
import jakarta.validation.Valid
import org.babyfish.jimmer.Page
import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.*

/**
 * OAuth2 注册客户端控制器
 *
 * 提供 OAuth2 注册客户端管理接口
 *
 * @author lishangbu
 * @since 2026/3/19
 */
@RestController
@RequestMapping("/oauth-registered-client")
class OauthRegisteredClientController(
    /** OAuth2 注册客户端服务 */
    private val oauthRegisteredClientService: OauthRegisteredClientService,
) {
    /**
     * 分页条件查询注册客户端
     *
     * @param pageable 分页参数
     * @return 注册客户端分页结果
     */
    @GetMapping("/page")
    fun getPage(
        pageable: Pageable,
        @ModelAttribute specification: OauthRegisteredClientSpecification,
    ): Page<OauthRegisteredClientView> = oauthRegisteredClientService.getPageByCondition(specification, pageable)

    /**
     * 条件查询注册客户端列表
     *
     * @return 注册客户端列表
     */
    @GetMapping("/list")
    fun list(
        @ModelAttribute specification: OauthRegisteredClientSpecification,
    ): List<OauthRegisteredClientView> = oauthRegisteredClientService.listByCondition(specification)

    /**
     * 根据 ID 查询注册客户端
     *
     * @param id 主键
     * @return 注册客户端
     */
    @GetMapping("/{id}")
    fun getById(
        @PathVariable id: String,
    ): OauthRegisteredClientView? = oauthRegisteredClientService.getById(id)

    /**
     * 新增注册客户端
     *
     * @param input 注册客户端写入请求
     * @return 保存后的注册客户端
     */
    @PostMapping
    fun save(
        @RequestBody @Valid input: SaveOauthRegisteredClientInput,
    ): OauthRegisteredClientView = oauthRegisteredClientService.save(input)

    /**
     * 更新注册客户端
     *
     * @param input 注册客户端写入请求
     * @return 更新后的注册客户端
     */
    @PutMapping
    fun update(
        @RequestBody @Valid input: UpdateOauthRegisteredClientInput,
    ): OauthRegisteredClientView = oauthRegisteredClientService.update(input)

    /**
     * 根据 ID 删除注册客户端
     *
     * @param id 主键
     */
    @DeleteMapping("/{id}")
    fun deleteById(
        @PathVariable id: String,
    ) {
        oauthRegisteredClientService.removeById(id)
    }
}
